syne_tune/tuner.py

```diff
@@ -46,7 +47,7 @@ def __init__(
             print_update_interval: float = 30.0,
             max_failures: int = 1,
             tuner_name: Optional[str] = None,
-            asynchronous_scheduling: bool = True,
+            asynchronous_scheduling: Optional[bool] = None,
```

Comment:
Why changing to Optional?

Reply:
The idea is that the default now depends on the scheduler. In most cases, it is True as before. But for schedulers which require synchronous, it would be False.

Reply:
I wrote this PR because both Aaron and me ran into this problem that if you want to run sync Hyperband and not set asynchronous_scheduling=False, you get pretty non-sense errors.

Reply:
Makes sense, thanks for moving this check in the caller. 

The important thing that I am trying to preserve is to not add any special case in the tuner regarding scheduler or backend as it would couple it to extra logic and make it hard to maintain.